applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -848,35 +847,50 @@ bool verify_datamode_control(uint16_t time_limit, uint16_t *min_time_limit)
 	return true;
 }
 
-struct at_param_list *slm_get_at_param_list(const char *at_cmd)
+int slm_get_at_param_list(const char *at_cmd, struct at_param_list *list)
 {
 	int err;
 
-	at_params_list_clear(&slm_at_param_list);
-	err = at_parser_params_from_str(at_cmd, NULL, &slm_at_param_list);
+	list = &at_host_param_list;
+
+	at_params_list_clear(list);
+	err = at_parser_params_from_str(at_cmd, NULL, list);
 	if (err) {
-		LOG_ERR("AT command parsing failed: %d", err);
-		at_params_list_clear(&slm_at_param_list);
+		LOG_ERR("AT command parse failed: %d", err);
+		at_params_list_clear(&at_host_param_list);
```

Comment:
Do we actually need to clear the list in case of failure? I am thinking that the case of the `%NCELLMEAS` notifications could be simplified by not clearing the list. Actually, now looking at what you have done there, this clearing definitely needs to be removed. Otherwise the notification parsing will try to proceed with an empty list.